### ISSUE TITLE
Promos - Use of Headings

### ISSUE THREAD

**Issue by greenc05** [bbc/gef-docs#42]
On Promos https://bbc.github.io/code-gel/components/promos/ the following contradicts the BBC Accessibility guideline advice on structure, which is correct BBC Guidelines or GEF? 

GEF: ‘**Must be marked up as an unordered list, with each Promo marked as a list item...Each Promo's primary link must be contained within a heading**’ 

https://www.bbc.co.uk/guidelines/futuremedia/accessibility/mobile/structure/headings

1. Headings guideline: Lists and headings used together to mark up content should be avoided;
2. Headings guideline: Headings should be coded around content that doesn’t change, i.e. a category name such as ‘UK News’ rather than a news item that will update frequently;
3. Headings guideline: Headings should not overwhelm a page as this can undermine their usefulness for screen reader users when navigating, a maximum of 7-9 headings should suffice.

**Comment by EmmaJP** [comment]
The current Promos doc states that groups of promos are marked up as a list, which means that heading elements are not needed within individual promo components. It also states that Promos should only have a single function, a link to the promoted content.

Is there ever a situation where there would be a single promo, and it might make sense for the promo headline to be a heading element? If not, then we could move to resolve this issue.

For example, I think the super promo thing at the top of the News page could be considered a different component from a standard promo.

**Comment by micmath** [comment]
+1 for the suggestion that card-like things with multiple links are not _typical_ Promos.

**Comment by greenc05** [comment]
@EmmaJP In the user research held recently on tabbed interfaces, we actually also included a promo in which we tested with and without a heading and it was clear cut that users prefer headings in promos.

**Comment by EmmaJP** [comment]
Definitely interested to see what was tested and the observations on that.

**Comment by JoshTumath** [comment]
I would also be interested to hear what has been found from user research. In conversations I've had with a couple of screen reader users, they normally don't like the large amount of headings on index pages that have promos. And we already tend to have a headings above each list of promos.

**Comment by greenc05** [comment]
@JoshTumath the research was carried out on an Article page, though I from observing the research and see the reasons why, I don't see why it wouldn't also apply to Index pages

**Comment by micmath** [comment]
_Summarising the gist of the discussion so far..._

**Proposed:** we recommend that each promo _should_ have their own Heading element

**For:**
> [HP] "I'm aware of just how much non-power users of screen readers rely on headings to get around."
> [CG] "We tested with and without a heading for the link, and it was clear cut that users prefer headings in promos."

**Against:**
> [JT] "Screen reader users... normally don't like the large amount of headings on index pages that have promos. And we already tend to have a headings above each list of promos."
> [EPR] "The promos themselves are links and the "headline" (as GEL calls it) is the link text. It rarely needs to be a heading element."

**Comment by micmath** [comment]
We need more data.

**Comment by micmath** [comment]
I've been reading the [user survey cited by @Heydon ](http://www.heydonworks.com/article/responses-to-the-screen-reader-strategy-survey) and possibly I misinterpreted the meaning of that in my previous comment, because it plainly does not say to  simply make more things headings; it actually argues for [_good document layout_ with _well-considered use of headings_](https://benrobertson.io/accessibility/designing-layouts-for-screen-readers). 

You could accomplish that by having a well-written heading followed by a list of promos which are essentially links.

```
[Hn]
    [li]
        [promo]
        [promo]
        [promo]
```

That changes the pro v con ratio to be in favour of not recommending that every promo have a heading.

@greenc05 would you agree that, considering all that, the advice should be essentially what MAG already says: "Use headings wisely to best describe the overall structure of the page to people who navigate by headings. Be aware that too many headings can make it hard to identify the main sections of your document."

**Comment by EmmaJP** [comment]
I remember that survey by Heydon. Some great insights in there.

I've asked Jack about this directly, and he is OK with lots of headings or lots of links. He's learned to navigate both and has never really compared the two experiences. That would be a great thing to get some research on ... comparing the two.

Watching the screen reader research iPlayer app is doing this week, there is a big difference in getting opinion on one way of doing things and then comparing that with a slightly different way of doing the same thing.

**Comment by greenc05** [comment]
@micmath I don't disagree with what you are saying, though it doesn't actually take into account how screen reader users actually navigate, and this is primarily via a heading short cut key from observed research (or the down arrow). Out of 8 screen reader users in the recent research I did not observe any navigating via the tab key... 

Do we have evidence to support:
"Screen reader users... normally don't like the large amount of headings on index pages that have promos. And we already tend to have a headings above each list of promos."? 

When @EmmaJP spoke to Jack this would suggest otherwise perhaps?

**Comment by micmath** [comment]
@greenc05 your question about evidence is addressed to Josh Tumath, since you are quoting him. If you read up the page you'll see that's based on talking to two users. Emma's comment was based on talking to one user, and your conclusion was based on a study with 8 users.

So far the only original thing I've said is I think we need more data. I still say that. But if we assume your study is indicative of most users, then the implication is that we should use headings in promos to aid navigation but also in many other patterns too.

But I'm loathe to start advising that since it contradicts what's in MAG. To quote you, from the original post of this thread:

- Headings guideline: Lists and headings used together to mark up content should be avoided;
- Headings guideline: Headings should be coded around content that doesn’t change, i.e. a category name such as ‘UK News’ rather than a news item that will update frequently;
- Headings guideline: Headings should not overwhelm a page as this can undermine their usefulness for screen reader users when navigating, a maximum of 7-9 headings should suffice.

Since GEF was created to be a companion to GEL and MAG, I think if we are to accept your conclusion, we need to fix MAG first.

**Comment by JoshTumath** [comment]
> Do we have evidence to support:
> "Screen reader users... normally don't like the large amount of headings on index pages that have promos. And we already tend to have a headings above each list of promos."?

To be honest, when I made that statement, I think it was really based on a combination of the testimony of one screen reader user I know and the exact guidelines/best practices that are outlined in the original post, so it's not worth much as an argument.

So I agree that we need more data to say whether using headings in promos is useful for screen readers. But I feel as though our default position for now should be to not use headings in promos, based on the guidelines noted above.

**Comment by micmath** [comment]
I think _all_ your statements are worth a lot to this conversation! <3

But there is a thing called "statistical significance" based on sample size, and certainly 1 or even 8 people isn't ideal.

**Comment by micmath** [comment]
In the absence of significant and convincing user data, we've always said we would revert to _standard industry practice_ so that we are at least [not being surprising](https://www.amazon.co.uk/Dont-Make-Think-Revisited-Usability/dp/0321965515/ref=dp_ob_image_bk).

We've already done a survey of [how it's done around the BBC](https://github.com/bbc/gef-docs/wiki/Promos) and in many cases teams are using: headings in lists, more than 9 headings per page, and headings for changable text. So there is prior practice for all of that at the BBC.

I've also looked around at other good websites:

The **homepage of gov.uk** for example has about 48 headings, many in lists, and many used for changable content.

**Barclays' website** uses headings in their promos:
```
<div class="promo-content ">
  <div class="promo-body">
    <h3>On your phone</h3>
    <p>Manage your money on your smartphone with our Barclays Mobile Banking app.</p>
  </div>
````

**Deque's website** use headings in lists and even use heading in their cards (a cousin pattern to the Promo):
```
<div class="m-card-body">
  <h3 class="m-card-title"> <span class="m-card-label">Free Tool</span> Calculate the Cost of False Positives</h3>
```
**The Guardian** has around 160 headings on its homepage, many in lists.

...

So I would argue that the advice in MAG about headings needs to be reconsidered. I've discussed this with Rebecca, who is managing changes to MAG at the moment and she has added that as a future task.

**Comment by JoshTumath** [comment]
There is another aspect to the argument against using headings in promos (other than accessibility, specifically), which is to do with what the HTML specification says a heading semantically is for:

> The first element of heading content in an element of sectioning content represents the heading for that section. Subsequent headings of equal or higher rank start new (implied) sections, headings of lower rank start implied subsections that are part of the previous one. In both cases, the element represents the heading of the implied section.

[(from the HTML spec)](https://html.spec.whatwg.org/multipage/sections.html#headings-and-sections)

If headings start 'implied sections of content', is each promo a section of content? If so, that means we could wrap a `section` element around each promo (but I don't think we actually would because that would spam the page with a lot of ARIA regions).

My personal opinion is that they are not because they are linking off to articles and there is no actual content within the promos themselves.

**Comment by micmath** [comment]
I don't think that spec is followed by websites so rigidly. When you mentioned sections you reminded me of the `<section>` tags I saw in the source for the Deque website (who we can assume, know their accessibility onions), and they actually follow a pattern very similar to what @greenc05 is suggesting we use for Promos:

```
<section>
    <div class="container">
        <h2 class="section-title">What you need to know about Accessibility</h2>
        
        <div class="section-description">
            <p>Accessibility is a multifaceted discipline and depending on your role and area of interest, what you need to know can differ. We’ve helped people from all skill and knowledge levels navigate their accessibility needs and would love to help you. Select your area of interest below to learn about which approach to accessibility might be best for you.</p>
        </div>

        <div class="call-out">
            <h3>Design</h3>
            <p>Taking an inclusive design approach can make an immediate user impact.</p>
            <a href="http://www.deque.com/accessible-design/">Read More <span class="sr-only">About Design Accessibility</span></a>
        </div>

        <div class="call-out">
            <h3>Compliance</h3>
            <p>Get Section 508, ADA, WCAG, AODA, and CVAA right the first time with our proven methodology.</p>
            <a href="http://www.deque.com/accessibility-compliance/" class="btn ghost gray">Read More <span class="sr-only">About Accessibility Compliance</span></a>
        </div>

    </div>
</section>
```
_* simplified for readability_

Where `H2` is used to describe the (literal) section, but then `H3`s are used to head the series of card-like components, which function _very much_ like our Promos.

**Comment by micmath** [comment]
@JoshTumath this is a line we agree on: "My personal opinion is that they are not [sections] because they are linking off to articles and there is no actual content within the promos themselves."

And I would be happy if our advice was something along the lines of: "If your promo is _merely_ a link to an article, then you should not use a Heading Element, but if you provide additional content, such as descriptive text, metadata, related links, etc, then use a Heading Element of one lower level than the section heading, in order to assist users who navigate primarily by heading"

Would that be agreeable to everyone?

**Comment by EmmaJP** [comment]
Of those who have implemented a basic promo with a heading, I'm curious how many have done so because they felt it was semantically correct or had tried it out with users, and how many did so because the design used bold text or the design pattern referred to it as a headline.

I do think cards that are in and of themselves a piece of content, should probably have a heading. I am much less inclined for a link to other content to be a heading, even if that link has some related metadata or an image informing a decision on which link to choose.

A set of promos could certainly be a section and should have a heading. The Deque example is closer to cards (IMHO) that are content themselves, with something akin to 'find out more' links.

For example:

**Heading for my list of links to stuff**

- Link to list item 1: a game
- Link to list item 2: a gallery
- Link to list item 3: a video

**Heading for my content card**
This is the content of my card. It summarises some stuff, can be read and makes sense in context of the page, and is complete as it is. 
And a link to find out more about stuff if you're interested.

**Comment by micmath** [comment]
> Of those who have implemented a basic promo with a heading, I'm curious how many have done so because they felt it was semantically correct or had tried it out with users, and how many did so because the design used bold text or the design pattern referred to it as a headline.

I don't think that applies to @greenc05 who has thought about semantics, understands the language in the pattern and has tried it out with users; and is now making a good argument to use the `Hn` element.

Your comment highlights two crucial points though:

1) Classification of  "a simple **card** with a link" versus "a **promo** with a bit of extra content" is on a continuum; There will be card-like promos and promo-like cards, depending on degrees. And sometimes it's a matter of context and perspective.

2) No one yet has citations of large-enough user studies that are convincing enough to make a recommendation obvious.

So, lacking 2, and with 1 making the identification of "industry practice" difficult, there isn't much left to say as a recommendation other than "consider your own use case and your own users carefully." Which, frankly, is what most teams around the BBC will say they are already doing.

I'm calling it then, and closing this thread until we can open a new one to consider new research data, when that is found or becomes available. (This will get easier when we bring people with the word _researcher_ in their job title into the mix.)
